config/eslint.config.js

```diff
@@ -169,23 +169,40 @@ module.exports = tseslint.config(
 				},
 			],
 
+			// strict
 			'@typescript-eslint/no-non-null-asserted-nullish-coalescing': 'error',
 			'@typescript-eslint/no-extraneous-class': 'error',
 			'@typescript-eslint/no-useless-constructor': 'error',
 			'@typescript-eslint/prefer-literal-enum-member': 'error',
 			'@typescript-eslint/no-invalid-void-type': 'error',
 
+			// recommended-type-checked
 			'@typescript-eslint/no-duplicate-type-constituents': 'error',
 			'@typescript-eslint/no-for-in-array': 'error',
 			'@typescript-eslint/no-implied-eval': 'error',
 			'@typescript-eslint/require-await': 'error',
 			'@typescript-eslint/await-thenable': 'error',
 			'@typescript-eslint/no-unnecessary-type-assertion': 'error',
 			'@typescript-eslint/no-redundant-type-constituents': 'error',
+			'@typescript-eslint/no-misused-promises': [
+				'error',
+				{
+					// It would add a lot of noise to React code b/c of passing async callbacks to elements
```

Comment:
So right now it just warns, right? Running lint in the UI, I don't see any warnings about this, so I'm not sure this will really be an issue

Reply:
No atm this rule is not enabled at all, since we just cherry pick individual rules from tseslint  `recommended-type-checked`  preset.
If we enable this, it will start complaining on any `onDone: (fn: (/*....*/) => void` prop that we might be passing in an async function, eg:
See: https://github.com/balena-io-modules/ui-shared-components/blob/b1894ada3b6cffda100af2f2f12bdd8bc74254ff/src/components/SimpleConfirmationDialog/index.tsx#L33-L35

Reply:
Makes sense 👍